src/aspire/source/micrograph.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+
+from aspire.image import Image
+from aspire.source import Simulation
+from aspire.source.image import _ImageAccessor
+
+
+class MicrographSource:
+    def __init__(
+        self,
+        micrograph_size=200,
+        L=300,
+        n=1,
+        ppm=100,
+        unique_filters=None,
+        dtype=None,
+        seed=0,
+        noise_adder=None,
+        collisions=False,
+        boundary=None,
+        interparticle_distance=None,
+    ):
+        """
+        A cryo-EM MicrographSource object that supplies micrographs.
+        :param micrograph_size: Size of micrograph
+        :param L: Image size for picked particle images
+        :param n: Number of micrographs to generate (integer)
+        :param ppm: Particles per micrograph
+        :param unique_filters: A list of Filter objects to be applied to micrograph
+        :param dtype: dtype for picked particle Simulation
+        :param seed: Random seed
+        :param noise_adder: Optionally append instance of NoiseAdder to generation pipeline
+        :param collisions: Optionally allow collisions
```

Comment:
I think if we have `interparticle_distance` then we no longer need the collisions, but we should describe the actions/defaults of `interparticle_distance` and `boundary`.